packages/ses/src/permits.js

```diff
@@ -1505,6 +1509,29 @@ export const permitted = {
     resolve: fn,
   },
 
+  // https://github.com/tc39/proposal-source-phase-imports?tab=readme-ov-file#js-module-source
+  '%AbstractModuleSourcePrototype%': {
+    constructor: '%AbstractModuleSource%',
+    '@@toStringTag': 'string',
+  },
+  '%AbstractModuleSource%': {
+    '[[Proto]]': '%FunctionPrototype%',
+    prototype: '%AbstractModuleSourcePrototype%',
+  },
```

Comment:
```suggestion
  '%AbstractModuleSource%': {
    '[[Proto]]': '%FunctionPrototype%',
    prototype: '%AbstractModuleSourcePrototype%',
  },

    '%AbstractModuleSourcePrototype%': {
    constructor: '%AbstractModuleSource%',
    '@@toStringTag': 'string',
  },

```

Please maintain the general pattern of the rest of the permits, where the `Foo` constructor comes before the `%FooPrototype%` prototype, and there is a blank line separating each permit.

Reply:
Likewise with `ModuleSource` below of course.

Reply:
Why does `%AbstractModuleSourcePrototype%` have a `@@toStringTag`? Is there precedent for other abstract superclasses having one, even though we always expect it to be overridden by concrete subclasses?

Reply:
The `@@toStringTag` is my error. There is no such property even on my shim. There is not precedent for abstract base classes among the intrinsics. No way to know yet whether the spec will land with one and it’s easy to add later.